views/default/icon/user/default.php

```diff
@@ -30,68 +30,48 @@
 $name = htmlspecialchars($user->getDisplayName(), ENT_QUOTES, 'UTF-8', false);
 $username = $user->username;
 
-$class = "elgg-avatar elgg-avatar-$size";
-if (isset($vars['class'])) {
-	$class = "$class {$vars['class']}";
-}
-if ($user->isBanned()) {
-	$class .= ' elgg-state-banned';
-	$banned_text = elgg_echo('banned');
-	$name .= " ($banned_text)";
-}
-
-$use_link = elgg_extract('use_link', $vars, true);
-
-$icontime = $user->icontime;
-if (!$icontime) {
-	$icontime = "default";
-}
+$wrapper_class = [
+	'elgg-avatar',
+	"elgg-avatar-$size",
+];
+$wrapper_class = elgg_extract_class($vars, $wrapper_class);
 
-$img_class = '';
-if (isset($vars['img_class'])) {
-	$img_class = elgg_extract('img_class', $vars);
-}
-
-
-$use_hover = elgg_extract('use_hover', $vars, true);
-if (isset($vars['hover'])) {
-	// only 1.8.0 was released with 'hover' as the key
-	$use_hover = elgg_extract('hover', $vars);
+if ($user->isBanned()) {
+	$wrapper_class[] = 'elgg-state-banned';
+	$name .= ' (' . elgg_echo('banned') . ')';
 }
 
 $icon = elgg_view('output/img', [
 	'src' => $user->getIconURL($size),
 	'alt' => $name,
 	'title' => $name,
-	'class' => $img_class,
+	'class' => elgg_extract('img_class', $vars), '',
```

Comment:
this should be `elgg_extract_class($vars, [], 'img_class);'`

ps: remove empty string at the end